internal/core/datahandle.cpp

```diff
@@ -207,6 +207,10 @@ DoubleDataHandle::DoubleDataHandle(OpenVDS::VDSHandle handle_a, OpenVDS::VDSHand
         this->m_binary_operator = &inplace_division;
     else
         throw detail::bad_request("Invalid binary_operator string");
+
+    if (this->m_metadata_a.crs().compare(this->m_metadata_b.crs()) != 0) {
```

Comment:
I think you can put it to the doublevolumedatalayout constructor, near the origin comparison and use layouts instead of the metadatas.

But if you are not in the mood, I can let it slip :)